src/documentation/documentationengine.cpp

```diff
@@ -27,13 +27,235 @@
+    constexpr const char* SettingsTitle = "Settings";
+    constexpr const char* SceneTitle = "Scene";
+    constexpr const char* PropertiesKeys = "properties";
+    constexpr const char* PropertyOwnersKey = "propertyOwners";
+    constexpr const char* TagsKey = "tags";
+    constexpr const char* UriKey = "uri";
+
+    // Scripting
+    constexpr const char* ScriptingTitle = "Scripting API";
+    constexpr const char* DefaultValueKey = "defaultValue";
+    constexpr const char* ArgumentsKey = "arguments";
+    constexpr const char* ReturnTypeKey = "returnType";
+    constexpr const char* HelpKey = "help";
+    constexpr const char* FileKey = "file";
+    constexpr const char* LineKey = "line";
+    constexpr const char* LibraryKey = "library";
+    constexpr const char* FullNameKey = "fullName";
+    constexpr const char* FunctionsKey = "functions";
+    constexpr const char* SourceLocationKey = "sourceLocation";
+    constexpr const char* OpenSpaceScriptingKey = "openspace";
+
+    // Licenses
+    constexpr const char* LicensesTitle = "Licenses";
+    constexpr const char* ProfileName = "Profile";
+    constexpr const char* AssetsName = "Assets";
+    constexpr const char* LicensesName = "Licenses";
+    constexpr const char* NoLicenseName = "No License";
+    constexpr const char* NoDataName = "";
+
+    constexpr const char* ProfileNameKey = "profileName";
+    constexpr const char* VersionKey = "version";
+    constexpr const char* AuthorKey = "author";
+    constexpr const char* UrlKey = "url";
+    constexpr const char* LicenseKey = "license";
+    constexpr const char* NoLicenseKey = "noLicense";
+    constexpr const char* IdentifiersKey = "identifiers";
+    constexpr const char* PathKey = "path";
+    constexpr const char* AssetKey = "assets";
+    constexpr const char* LicensesKey = "licenses";
+
+    // Keybindings
+    constexpr const char* KeybindingsTitle = "Keybindings";
+    constexpr const char* KeybindingsKey = "keybindings";
+
+    // Events
+    constexpr const char* EventsTitle = "Events";
+    constexpr const char* FiltersKey = "filters";
+    constexpr const char* ActionsKey = "actions";
+
+    nlohmann::json documentationToJson(
+                                         const openspace::documentation::Documentation& d)
+    {
+        using namespace openspace::documentation;
+
+        nlohmann::json json;
+
+        json[NameKey] = d.name;
+        json[IdentifierKey] = d.id;
+        json[DescriptionKey] = d.description;
+        json[MembersKey] = nlohmann::json::array();
+
+        for (const DocumentationEntry& p : d.entries) {
+            nlohmann::json entry;
+            entry[NameKey] = p.key;
+            entry[OptionalKey] = p.optional.value;
+            entry[TypeKey] = p.verifier->type();
+            entry[DocumentationKey] = p.documentation;
+
+            auto* tv = dynamic_cast<TableVerifier*>(p.verifier.get());
+            auto* rv = dynamic_cast<ReferencingVerifier*>(p.verifier.get());
```

Comment:
```suggestion
            ReferencingVerifier* rv = dynamic_cast<ReferencingVerifier*>(p.verifier.get());
```

Reply:
We _did_ add the exception for `auto` if the type is explicitly mentioned on the right side